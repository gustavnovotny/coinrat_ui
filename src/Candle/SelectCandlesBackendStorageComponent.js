// @flow
import React from "react"
import {SelectComponent} from "../SelectComponent"
import type {DefaultSelectDataType, SelectElement} from "../SelectComponent"

type Props = {
  availableStorages: DefaultSelectDataType,
  onSelect: (candleStorage: string) => void,
  defaultSelectedPair: string,
}

const SelectCandlesBackendStorageComponent = ({availableStorages, defaultSelectedCandleStorage, onSelect}: Props) => {
  const item = availableStorages[defaultSelectedCandleStorage]
  return <SelectComponent
    label="Candle storage"
    items={availableStorages}
    selectedItem={item}
    onChange={(candleStorage: SelectElement) => onSelect(candleStorage.key)}
  />
}

export default SelectCandlesBackendStorageComponent
