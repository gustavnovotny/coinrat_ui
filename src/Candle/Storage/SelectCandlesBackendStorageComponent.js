// @flow
import React from "react"
import {SelectItemsType, SelectElement, SelectComponent} from "../../Form/Select/SelectComponent"

type Props = {
  availableStorages: SelectItemsType,
  onSelect: (candleStorage: string) => void,
  defaultSelectedPair: string,
}

const SelectCandlesBackendStorageComponent = ({availableStorages, defaultSelectedCandleStorage, onSelect}: Props) => {
  const item = availableStorages[defaultSelectedCandleStorage]
  return <SelectComponent
    label="Candle storage"
    items={availableStorages}
    selectedItem={item || null}
    onChange={(candleStorage: SelectElement) => onSelect(candleStorage.key)}
  />
}

export default SelectCandlesBackendStorageComponent
