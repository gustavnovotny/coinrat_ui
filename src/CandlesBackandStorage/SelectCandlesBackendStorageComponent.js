import React, {Component} from "react"
import {observer} from "mobx-react/index"
import SelectComponent from "../SelectComponent"

const SelectCandlesBackendStorageComponent = observer(class SelectPairComponent extends Component {

  handleValueChange = (pair) => this.props.store.changeSelectedCandleStorage(pair.key)

  render() {
    const candleBackendStorages = {
      'influx_db': {key: 'influx_db', title: 'Influx DB'},
      'memory': {key: 'memory', title: 'In memory'},
    }

    const item = candleBackendStorages[this.props.store.selectedCandleStorage]

    return <SelectComponent
      label="Candle storage"
      items={candleBackendStorages}
      selectedItem={item}
      onChange={this.handleValueChange}
    />
  }
})

export default SelectCandlesBackendStorageComponent
