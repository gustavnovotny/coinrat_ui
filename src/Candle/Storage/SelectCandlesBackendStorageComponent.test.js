// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import SelectCandlesBackendStorageComponent from "./SelectCandlesBackendStorageComponent"

it('renders correctly', () => {
  const onSelect = () => undefined
  const tree = renderer
    .create(
      <SelectCandlesBackendStorageComponent
        availableStorages={{foo: {key: 'foo', title: 'Foo'}}}
        defaultSelectedPair="foo"
        onSelect={onSelect}
      />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
