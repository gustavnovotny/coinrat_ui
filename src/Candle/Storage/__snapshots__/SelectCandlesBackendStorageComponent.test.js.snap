// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="_Rfx0_"
>
  <div
    className="form-item form-label _Rfx1_"
  >
    <div
      className="pt-label"
    >
      Candle storage:
      <span
        className="pt-text-muted"
      />
    </div>
  </div>
  <div
    className="form-item form-element"
  >
    <span>
      <div
        className="pt-popover-target"
        onClick={[Function]}
      >
        <div
          className=""
          disabled={undefined}
          onKeyDown={[Function]}
          onKeyUp={undefined}
          tabIndex={undefined}
        >
          <button
            className="pt-button"
            disabled={undefined}
            onClick={undefined}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <div
              className="_Rfx2_ _Rfx0_"
            >
              <div
                className=""
                style={
                  Object {
                    "whiteSpace": "nowrap",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "marginBottom": "-5px",
                      "marginTop": "5px",
                    }
                  }
                >
                  <div
                    className="pt-spinner pt-small"
                  >
                    <div
                      className="pt-spinner-svg-container"
                    >
                      <svg
                        viewBox="-15 -15 130 130"
                      >
                        <path
                          className="pt-spinner-track"
                          d="M 50,50 m 0,-44.5 a 44.5,44.5 0 1 1 0,89 a 44.5,44.5 0 1 1 0,-89"
                        />
                        <path
                          className="pt-spinner-head"
                          d="M 50,50 m 0,-44.5 a 44.5,44.5 0 1 1 0,89 a 44.5,44.5 0 1 1 0,-89"
                          pathLength={280}
                          style={
                            Object {
                              "strokeDasharray": "280 280",
                              "strokeDashoffset": 210,
                            }
                          }
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=""
              >
                <span
                  className="pt-icon-standard pt-icon-caret-down"
                />
              </div>
            </div>
          </button>
        </div>
      </div>
    </span>
  </div>
  <div
    className="form-item form-suffix"
  >
    
     
    
  </div>
</div>
`;
